<?xml version="1.0"?>
<launch>

  <node pkg="robot_localization" type="ekf_localization_node" name="rover_odom_localization">
    <rosparam command="load" file="$(find rover_autonomy)/config/rover_odom_localization.yaml" />
    <remap from="/odometry/filtered" to="/odometry/rover_odom" />  
  </node>

  <node pkg="robot_localization" type="ekf_localization_node" name="rover_gps_odom_localization">
    <rosparam command="load" file="$(find rover_autonomy)/config/rover_gps_odom_localization.yaml" />
    <remap from="/odometry/filtered" to="/odometry/rover_gps_odom" />
  </node>

  <node pkg="robot_localization" type="navsat_transform_node" name="rover_navsat_transform" respawn="true">
    <!-- <param name="magnetic_declination_radians" value="-9.652"/>  -->
     <param name="magnetic_declination_radians" value="-10.715"/>
    <param name="yaw_offset" value="3.14159"/>
    <param name="zero_altitude" value="true"/>
    <param name="broadcast_utm_transform" value="true"/>

    <param name="publish_filtered_gps" value="true"/>
    <!-- input mapping -->
    <!--remap from="/imu/data" to=""/-->
    <remap from="/gps/fix" to="/navsat/fix"/>
    <remap from="/odometry/filtered" to="/odometry/rover_gps_odom"/> <!-- it is looking for /odometry/filtered, we want /odometry/rover_gps_odom-->
    
    <!-- output mapping -->
    <remap from="/odometry/gps" to="/odometry/rover_gps"/>

  </node>

</launch>

